frontend/src/reviewRoles/ReviewRoleFormContainer.tsx

```diff
@@ -0,0 +1,277 @@
+import EntityIcon from 'src/entry/EntityIcon'
+import EntityNameDisplay from 'src/entry/EntityNameDisplay'
+import EntryAccessInput from 'src/entry/settings/EntryAccessInput'
+import MessageAlert from 'src/MessageAlert'
+import { KeyedMutator } from 'swr'
+import { CollaboratorEntry, EntryKind, ReviewRoleInterface, ReviewRolesFormData, SystemRoleKeys } from 'types/types'
+import { getErrorMessage } from 'utils/fetcher'
+
+type ReviewRolesFormDataUnion = UpdateReviewRolesParams | ReviewRolesFormData
+
+type ReviewRoleFormContainerProps = {
+  providedData: boolean
+  formData?: ReviewRolesFormDataUnion
+  setFormData: (prevFormData: any) => void
+  setIsEdit?: (state: boolean) => void
+  mutateReviewRoles?: KeyedMutator<{
+    reviewRoles: ReviewRoleInterface[]
+  }>
+  headingComponent: ReactElement
+}
+
+export default function ReviewRoleFormContainer({
+  providedData = false,
+  formData = { name: '', shortName: '', description: '', systemRole: 'none', defaultEntities: [] },
+  setFormData,
+  setIsEdit = () => {},
+  mutateReviewRoles,
+  headingComponent,
+}: ReviewRoleFormContainerProps) {
+  const [defaultEntitiesEntry, setDefaultEntities] = useState<Array<CollaboratorEntry>>(
+    formData.defaultEntities
+      ? formData.defaultEntities.map((defaultEntity) => ({ entity: defaultEntity, roles: [] }))
+      : [{ entity: '', roles: [] }],
+  )
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles()
+
+  const [errorMessage, setErrorMessage] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const handleNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setFormData((prevFormData: UpdateReviewRolesParams) => ({ ...prevFormData, name: event.target.value as string }))
+  }
+
+  const handleShortNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setFormData((prevFormData: UpdateReviewRolesParams) => ({
+      ...prevFormData,
+      shortName: event.target.value as string,
+    }))
+  }
+
+  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setFormData((prevFormData: UpdateReviewRolesParams) => ({
+      ...prevFormData,
+      description: event.target.value as string,
+    }))
+  }
+
+  const handleSystemRoleChange = (event: SelectChangeEvent) => {
+    if (event.target.value === '') {
+      delete formData.systemRole
+    } else {
+      setFormData((prevFormData: UpdateReviewRolesParams) => ({
+        ...prevFormData,
+        systemRole: event.target.value.toLowerCase() as SystemRoleKeys,
+      }))
+    }
+  }
+
+  const handleDefaultEntitiesChange = useMemo(() => {
+    return (newValue: Array<CollaboratorEntry>) => {
+      setDefaultEntities(newValue)
+    }
+  }, [])
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    setErrorMessage('')
+    setLoading(true)
+
+    if (providedData) {
+      //Existing data (view.tsx)
+
+      const res = await putReviewRole({
```

Comment:
If you move the logic for posting/putting the data inside the parent(s) instead then you won't need this conditional check in here.